app/code/community/Algolia/Algoliasearch/Model/Resource/Fulltext/Collection.php

```diff
@@ -3,17 +3,50 @@
 class Algolia_Algoliasearch_Model_Resource_Fulltext_Collection extends Mage_CatalogSearch_Model_Resource_Fulltext_Collection
 {
     /**
-     * Intercept query.
+     * Get found products ids
+     *
+     * @return array|Algolia_Algoliasearch_Model_Resource_Fulltext_Collection
      */
-    public function addSearchFilter($query)
+    public function getFoundIds()
     {
-        /** @var Algolia_Algoliasearch_Helper_Config $config */
-        $config = Mage::helper('algoliasearch/config');
+        if (!$this->_helper()->isX3Version()) {
+            return $this;
+        }
+        
+        $query = $this->_getQuery();
+        if (is_null($this->_foundData) && !empty($query)) {
+            $data = $this->getAlgoliaData($query);
```

Comment:
`$query` variable here is not a string, but instance of `Mage_CatalogSearch_Model_Query`. And it causes issues and does not retrieves correct data from Algolia because search string is then empty in real search function. I fixed it on my side by passing `$query->query_text` instead of just `$query`.

Can you fix it @lukaszqs? Thanks!